app/src/components/market/market_home.tsx

```diff
@@ -0,0 +1,64 @@
+import React, { useState, useEffect } from 'react'
+import styled from 'styled-components'
+
+import { MarketAndQuestion, MarketFilters, Status } from '../../util/types'
+import { FullLoading } from '../common/full_loading'
+import { ListCard } from '../common/list_card'
+import { ListItem } from '../common/list_item'
+import { SectionTitle } from '../common/section_title'
+import { Filter } from '../common/filter'
+import { ConnectedWeb3Context } from '../../hooks/connectedWeb3'
+import { DisconnectedWeb3Context } from '../../hooks/disconnectedWeb3'
+
+const FilterStyled = styled(Filter)`
+  margin: -30px auto 10px;
+  max-width: ${props => props.theme.list.maxWidth};
+  width: 100%;
+`
+
+interface Props {
+  markets: MarketAndQuestion[]
+  status: Status
+  context: ConnectedWeb3Context | DisconnectedWeb3Context
+}
+
+export const MarketHome: React.FC<Props> = (props: Props) => {
+  const { status, markets, context } = props
+  const options = [MarketFilters.AllMarkets, MarketFilters.MyMarkets]
+  const defaultOption = options[0]
+
+  const [marketsFiltered, setMarketsFiltered] = useState<MarketAndQuestion[]>([])
+
+  useEffect(() => {
+    setMarketsFiltered(markets)
+  }, [markets])
+
+  const filterMarkets = (event: { name: string; value: string }) => {
```

Comment:
Several comments here:

1. `name` is not used
2. Receiving an event is ugly; it's better to extract the value in a lambda function and to call the method with it.
3. I think all of this should be done in a `useEffect` whose dependencies are `markets` and something like `currentFilter`.

Reply:
I update all this mechanism to use a hooks :+1: